File: src/main/java/com/github/kindrat/cassandra/client/ui/listener/TextChangeListener.java
package com.github.kindrat.cassandra.client.ui.listener;

import javafx.beans.value.ChangeListener;
import javafx.beans.value.ObservableValue;
import lombok.RequiredArgsConstructor;

import java.util.Objects;
import java.util.function.Consumer;

@RequiredArgsConstructor
public class TextChangeListener implements ChangeListener<String> {
    private final Consumer<String> action;

    @Override
    public void changed(ObservableValue<? extends String> observable, String oldValue, String newValue) {
        if (!Objects.equals(oldValue, newValue)) {
            action.accept(newValue);
        }
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/ui/listener/TextChangeListener.kt
package com.github.kindrat.cassandra.client.ui.listener

import javafx.beans.value.ChangeListener
import javafx.beans.value.ObservableValue
import java.util.function.Consumer

class TextChangeListener(private val action: Consumer<String>) : ChangeListener<String> {

    override fun changed(observable: ObservableValue<out String>, oldValue: String, newValue: String) {
        if (oldValue != newValue) {
            action.accept(newValue)
        }
    }

}